performance/kafka_rw/main.go

```diff
@@ -55,40 +56,38 @@ func main() {
 	for i := 0; i < topics; i++ {
 		topic := nuid.Next()
 		log.Printf("creating topic %s", topic)
-		connection, err := kafka.DialContext(context.Background(), "tcp", kafkaHostPort)
-		if connection == nil || err != nil {
+		connection, err := kafka.NewManager(conf.ConnectorConfig{
+			Brokers: []string{kafkaHostPort},
+		}, 5*time.Second)
+		if err != nil {
 			log.Fatalf("unable to connect to kafka server")
 		}
 
-		connection.SetDeadline(time.Now().Add(15 * time.Second))
-		err = connection.CreateTopics(kafka.TopicConfig{
-			Topic:             topic,
-			NumPartitions:     1,
-			ReplicationFactor: 1,
-		})
+		err = connection.CreateTopic(topic, 1, 1)
 		connection.Close()
 		if err != nil {
 			log.Fatalf("error creating topic, %s", err.Error())
 		}
 
 		// start the writer
 		go func(topic string) {
-			writer := kafka.NewWriter(kafka.WriterConfig{
-				Brokers:  []string{kafkaHostPort},
-				Topic:    topic,
-				Balancer: &kafka.LeastBytes{},
-			})
+			writer, err := kafka.NewProducer(conf.ConnectorConfig{
+				Brokers: []string{kafkaHostPort},
+				Topic:   topic,
+			}, 5*time.Second, topic)
```

Comment:
Would it make sense to change the `5*time.Second` to something like `conn.bridge.config.ConnectTimeout` so that it is configurable? I see that we have the 5sec timeout in a few places now

Reply:
Would be good to avoid hardcoded timeouts here.  All Kafka timeouts could be set as a parameter/options and probably use very generous defaults because Kafka can be so slow in some deployments.